iceoryx_posh/source/popo/gateway_generic.cpp

```diff
@@ -28,15 +28,25 @@ GatewayGeneric::GatewayGeneric(const capro::Interfaces f_interface) noexcept
 
 GatewayGeneric::~GatewayGeneric() noexcept
 {
-    if(m_interfaceImpl)
+    if (m_interfaceImpl)
     {
         m_interfaceImpl.destroy();
     }
 }
 
 bool GatewayGeneric::getCaProMessage(CaproMessage& msg) noexcept
```

Comment:
I would suggest changing the signature to `cxx::optional<CaproMessage> GatewayGeneric::getCaProMessage() noexcept`
This would then be consistent to all the other methods which can return a value but do not have to.
Additionally, all the other ports have this signature.

Reply:
This is no more port but the next layer. Here we have a new game and breaking this API has consequences for all legacy gateways. The costs are high for having it more the modern C++ way. And it is not inconsistent as it is another layer...